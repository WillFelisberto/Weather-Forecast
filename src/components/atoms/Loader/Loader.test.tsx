import { render, screen } from '@/tests/test-utils';

import { Loader } from '.';

describe('Loader Component', () => {
  it('renders without crashing', () => {
    const { container } = render(<Loader />);
    expect(screen.getByRole('status')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('displays the passed text', () => {
    const text = 'Loading data';
    render(<Loader text={text} />);
    expect(screen.getByText(text)).toBeInTheDocument();
  });
});
