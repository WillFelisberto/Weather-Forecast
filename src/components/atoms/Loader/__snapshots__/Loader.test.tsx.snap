// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders without crashing 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center pt-8 "
    role="status"
  >
    <svg
      class="lucide lucide-loader-circle animate-spin animate-pulse text-gray-500"
      fill="none"
      height="50"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
  </div>
</div>
`;
